core/services/assets/AssetManifest.php

```diff
@@ -0,0 +1,329 @@
+    private $assets_namespace;
+
+    /**
+     * @var string
+     */
+    private $assets_path;
+
+    /**
+     * @var DomainInterface
+     */
+    protected $domain;
+
+    /**
+     * @var array
+     */
+    private $entry_points;
+
+    /**
+     * @var array
+     */
+    private $manifest;
+
+    /**
+     * @var string
+     */
+    private $manifest_path;
+
+    /**
+     * This is a list of known handles that are used for css.
+     * @var array
+     */
+    private $wp_css_handle_dependencies = [
+        'wp-components',
+        'wp-block-editor',
+        'wp-block-library',
+        'wp-edit-post',
+        'wp-edit-widgets',
+        'wp-editor',
+        'wp-format-library',
+        'wp-list-reusable-blocks',
+        'wp-nux',
+    ];
+
+
+    /**
+     * AssetManifest constructor.
+     *
+     * @param DomainInterface $domain
+     */
+    public function __construct(DomainInterface $domain)
+    {
+        $this->domain = $domain;
+    }
+
+
+    /**
+     * @return void
+     */
+    public function initialize()
+    {
+        if (! $this->manifest) {
+            $this->assets_namespace = $this->domain->assetNamespace();
+            $assets_path = $this->domain->distributionAssetsPath();
+            $this->assets_path = trailingslashit($assets_path);
+            $this->setManifestFilepath();
+            $this->loadManifest();
+            $this->getAssetFiles();
+            $this->getEntryPoints();
+        }
+    }
+
+
+    /**
+     * @param string $manifest_path
+     */
+    public function setManifestFilepath($manifest_path = '')
+    {
+        $manifest_path = $manifest_path ? $manifest_path : $this->assets_path . AssetManifest::FILE_NAME;
+        if (! is_readable($manifest_path)) {
+            throw new AssetManifestException(
+                sprintf(
+                    esc_html__(
+                        'Manifest file was not found or is not readable. Please try running "yarn dev" in the console and/or verify that "%1$s" exists and has valid permissions.',
+                        'event_espresso'
+                    ),
+                    $manifest_path
+                )
+            );
+        }
+        $this->manifest_path = $manifest_path;
+    }
+
+
+    /**
+     * @return void
+     */
+    private function loadManifest()
+    {
+        if (! $this->manifest) {
+            // TODO May be use WP File system? ¯\_(ツ)_/¯
+            $manifest_json  = file_get_contents($this->manifest_path);
+            $this->manifest = json_decode($manifest_json, true);
+        }
+    }
+
+
+    /**
+     * @param string $file_name
+     * @return string
+     */
+    private function trimAssetFilename($file_name)
+    {
+        return ltrim($file_name, '/');
+    }
+
+
+    /**
+     * @return array
+     */
+    public function getAssetFiles()
+    {
+        if (! $this->asset_files) {
+            if (empty($this->manifest[ AssetManifest::KEY_FILES ])) {
+                if (WP_DEBUG) {
```

Comment:
```suggestion
                if (defined('WP_DEBUG') && WP_DEBUG) {
```

Reply:
from `wp-includes/default-constants.php`

```php
	if ( ! defined( 'WP_DEBUG' ) ) {
		if ( 'development' === wp_get_environment_type() ) {
			define( 'WP_DEBUG', true );
		} else {
			define( 'WP_DEBUG', false );
		}
	}
```

it's always defined

the one constant you DO always have to check for is `DOING_AJAX`

Reply:
Aha, thanks, TIL